docs/nsm_api.md

```diff
@@ -0,0 +1,263 @@
+Network Service Mesh (NSM) API
+
+## Executive summary
+
+Based on the discussions on NSM IRC channel on September 11th/12th and in the issue [https://github.com/ligato/networkservicemesh/issues/283](https://github.com/ligato/networkservicemesh/issues/283). This document describes API calls between different components in NSM enabled kubernetes cluster. Each type of API is provided with proto file definition and brief description of parameters expected and returned. Developers of NFVi who plan to leverage NSM for their applications, are urged to thoroughly review these API calls to make sure that expected and returned parameters do fully cover their applications&#39; needs.
+
```

Comment:
As @iawells indicated, it would be good to add a section to this doc (or a different markdown file, referenced here) with usage examples of the APIs defined here.

Reply:
@mestery Could you please clarify what you mean by usage examples? Do you mean code examples  or you mean something like a walk through: NSM client talks to its local NSM etc?